tf2/include/tf2/convert.h

```diff
@@ -137,6 +157,48 @@ void convert(const A & a1, A & a2)
     a2 = a1;
   }
 }
+
+/**\brief Function that converts from a row-major representation of a 6x6
+ * covariance matrix to a nested array representation.
+ * \param row_major A row-major array of 36 covariance values.
+ * \param nested_array A nested array representation of 6x6 covariance values.
+ */
+inline
+std::array<std::array<double, 6>, 6> convertCovariance(const std::array<double, 36> & row_major)
```

Comment:
I don't love the name `convertCovariance`, particularly in combination with the version below.  It just seems too hard to figure out what it does from the name.  I'd prefer something more like:

```
covarianceRowMajorToNested()
```

And then below:

```
covarianceNestedToRowMajor()
```

Reply:
Changed in 34246f8.